modules/catalog-magento/services/useGetCategoryById.ts

```diff
@@ -0,0 +1,36 @@
+import useGetCategoryByIdRepository, { CategoryOptions } from '#ioc/repositories/useGetCategoryByIdRepository'
+import ToProduct from '#ioc/mappers/ToProduct'
+import IS_CLIENT from '#ioc/config/IS_CLIENT'
+
+export default () => {
+  const getCategoryByIdRepository = useGetCategoryByIdRepository()
+  let prevGetProducts = [] as ReturnType<typeof ToProduct>[]
+  let storedPageSize = false
+
+  return async (...args: CategoryOptions) => {
```

Comment:
Copy paste the type definition for arguments here, They can differe from arguments to repository.